src/services/uploadFile.js

```diff
@@ -14,63 +14,64 @@ const acceptedFileTypes = []; // allow all
         const fileName = path.basename(filePath);
 
         // Validate the file before proceeding
         if (!validateUploadFile(filePath)) {
             updateStatusBar({ mainWindow, text: `Invalid file: ${fileName}. Reason: ${validationError}`, type: 'danger' });
-            continue; // Skip to the next file
+            return null; // Skip to the next file
         }
 
         const moveFilePending = path.basename(filePath);
         const newFilePath = path.join(pendingUploadsPath, moveFilePending);
         fsExtra.renameSync(filePath, newFilePath); // 
-
-    }
+        return newFilePath;
+    }).filter((path) => path !== null);
 
     try {
-        await uploadFilesFTP(mainWindow, filePathsCopy, basePath);
+        await uploadFilesFTP(mainWindow, pendingFilePaths, basePath);
         updateStatusBar({ mainWindow, text: `Uploaded files: ${new Date()}`, type: 'success' });
     } catch (error) {
         updateStatusBar({ mainWindow, text: `Error uploading files: ${error}`, type: 'danger' });
     }
 };
 
 const uploadFilesFTP = async (mainWindow, filePaths, basePath) => {
-
     const ftpDetails = await getSftpDetails();
     const { ftpHost, ftpUsername, ftpPassword, ftpPort, ftpTargetPath } = ftpDetails;
 
     await connect({ host: ftpHost, username: ftpUsername, password: ftpPassword, port: ftpPort });
 
-    await Promise.allSettled(filePaths.map(async (filePath) => {
-        const fileName = path.basename(filePath);
-        const targetPath = `${ftpTargetPath}/${fileName}`;
-        const basePathParts = basePath.split('/');
-        basePathParts.pop();
-        let newBasePath = basePathParts.join('/') + '/';
-
-        const lastIndex = filePath.lastIndexOf('/');
-        const fileNameEnding = filePath.substring(lastIndex + 1);
-
-        try {
-            await getClient().fastPut(`${newBasePath}ftp_files_pending/${fileNameEnding}`, targetPath);
-            updateStatusBar({ mainWindow, text: `File Uploaded: ${fileNameEnding}`, type: 'success' });
-        } catch (error) {
-            logError(error, `Error uploading file "${fileName}" to FTP server: ${error.message}`);
-            throw error;
-        }
-    }));
-
-    var lastIndex = basePath.lastIndexOf('/');
+    await Promise.all(
+        filePaths.map(async (filePath) => {
+            const fileName = path.basename(filePath);
+
+            
+            const targetPath = path.join(ftpTargetPath, fileName);
+            
+            const lastIndex = filePath.lastIndexOf(path.sep);
+            const fileNameEnding = filePath.substring(lastIndex + 1);
+            try {
+                await getClient().fastPut(filePath, targetPath);
+                updateStatusBar({ mainWindow, text: `File Uploaded: ${fileNameEnding}`, type: 'success' });
+            } catch (error) {
+                logError(error, `Error uploading file "${fileName}" to FTP server: ${error.message}`);
+                throw error;
+            }
+        })
+    );
```

Comment:
This is very bad syntax, and not the right way to use Promise.all but it's working.... 

probably should be tidied up at some point